In [1]:
# ライブラリのインポート
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# 元となるデータの取得
df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-5/user-5-nonull.csv")
df_origin.head()

/tmp/ipython-input-2-1337201291.py:2: DtypeWarning: Columns (68) have mixed types. Specify dtype option on import or set low_memory=False.
  df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-5/user-5-nonull.csv")


Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0           327258000          NaN   
1           1                 9016           327267016        Mouse   
2           2                14166           327272166  Eye Tracker   
3           3                24233           327282233        Mouse   
4           4                31285           327289285  Eye Tracker   

  Project name Export date Participant name Recording name Recording date  \
0          JD4  2025/07/18           user-5     Recording6     2025/07/18   
1          JD4  2025/07/18           user-5     Recording6     2025/07/18   
2          JD4  2025/07/18           user-5     Recording6     2025/07/18   
3          JD4  2025/07/18           user-5     Recording6     2025/07/18   
4          JD4  2025/07/18           user-5     Recording6     2025/07/18   

  Recording date UTC  ... Original Media height Eye movement type  \
0         2025/07/18  ...                   NaN               NaN   
1         2025/07/18  ...                   NaN      EyesNotFound   
2         2025/07/18  ...                   NaN      EyesNotFound   
3         2025/07/18  ...                   NaN      EyesNotFound   
4         2025/07/18  ...                   NaN      EyesNotFound   

   Eye movement event duration Eye movement type index Fixation point X  \
0                          NaN                     NaN              NaN   
1                      16889.0                     1.0              NaN   
2                      16889.0                     1.0              NaN   
3                      16889.0                     1.0              NaN   
4                      16889.0                     1.0              NaN   

  Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  \
0              NaN                         NaN                         NaN   
1              NaN                         NaN                         NaN   
2              NaN                         NaN                         NaN   
3              NaN                         NaN                         NaN   
4              NaN                         NaN                         NaN   

   Mouse position X  Mouse position Y  
0               NaN               NaN  
1            1633.0             667.0  
2               NaN               NaN  
3            1624.0             658.0  
4               NaN               NaN  

[5 rows x 84 columns]

# カラムの整理

In [3]:
nunique_series = df_origin.nunique(dropna=False).to_string()
print(nunique_series)

Unnamed: 0                                    17492
Recording timestamp                           17479
Computer timestamp                            17479
Sensor                                            3
Project name                                      1
Export date                                       1
Participant name                                  1
Recording name                                    1
Recording date                                    1
Recording date UTC                                1
Recording start time                              1
Recording start time UTC                          1
Recording duration                                1
Timeline name                                     1
Recording Fixation filter name                    1
Recording software version                        1
Recording resolution height                       1
Recording resolution width                        1
Recording monitor latency                         1
Average cali

In [4]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series = df_origin.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

In [5]:
# カラム名とそのユニーク値を出力
for col in single_unique_cols:
    unique_val = df_origin[col].unique()
    print(f"{col}: {unique_val}")


Project name: ['JD4']
Export date: ['2025/07/18']
Participant name: ['user-5']
Recording name: ['Recording6']
Recording date: ['2025/07/18']
Recording date UTC: ['2025/07/18']
Recording start time: ['16:34:24.211']
Recording start time UTC: ['07:34:24.211']
Recording duration: [309213]
Timeline name: ['Timeline1']
Recording Fixation filter name: ['Tobii I-VT (Fixation)']
Recording software version: ['24.21.435']
Recording resolution height: [1080]
Recording resolution width: [1920]
Recording monitor latency: [10]
Average calibration accuracy (mm): [14.7]
Average calibration accuracy (degrees): [1.12]
Average calibration accuracy (pixels): [51]
Average validation accuracy (mm): [11]
Average validation precision SD (mm): [4.1]
Average validation precision RMS (mm): [5.5]
Average validation accuracy (degrees): [0.84]
Average validation precision SD (degrees): [0.32]
Average validation precision RMS (degrees): [0.43]
Average validation accuracy (pixels): [38]
Average validation precision S

## カラムの要素が1つしかない変数の説明
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-5`|
|Recording name| 録画名|`Recording6`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:34:24.211`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:34:24.211`|
|Recording duration| 録画時間|`309213`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`14.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.12`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`51`pixels|
|Average validation accuracy | 平均バリデーション精度|`11`mm|
|Average validation precision SD | 平均バリデーション予測SD|`4.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`5.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.84`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.32`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.43`degrees|
|Average validation accuracy |平均バリデーション精度| `38`pixels|
|Average validation precision SD |平均バリデーション予測SD| `14`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `19`pixels|

In [6]:
# 削除した新しいデータフレーム
df_cleaned = df_origin.drop(columns=single_unique_cols)
df_cleaned.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0           327258000          NaN   
1           1                 9016           327267016        Mouse   
2           2                14166           327272166  Eye Tracker   
3           3                24233           327282233        Mouse   
4           4                31285           327289285  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1                   NaN             NaN         NaN           NaN   
2           342651795.0             NaN         NaN           NaN   
3                   NaN             NaN         NaN           NaN   
4           342668913.0             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Original Media height  \
0           NaN                NaN  ...                    NaN   
1           NaN                NaN  ...                    NaN   
2           NaN                NaN  ...                    NaN   
3           NaN                NaN  ...                    NaN   
4           NaN                NaN  ...                    NaN   

   Eye movement type  Eye movement event duration  Eye movement type index  \
0                NaN                          NaN                      NaN   
1       EyesNotFound                      16889.0                      1.0   
2       EyesNotFound                      16889.0                      1.0   
3       EyesNotFound                      16889.0                      1.0   
4       EyesNotFound                      16889.0                      1.0   

   Fixation point X  Fixation point Y  Fixation point X (MCSnorm)  \
0               NaN               NaN                         NaN   
1               NaN               NaN                         NaN   
2               NaN               NaN                         NaN   
3               NaN               NaN                         NaN   
4               NaN               NaN                         NaN   

   Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
0                         NaN               NaN               NaN  
1                         NaN            1633.0             667.0  
2                         NaN               NaN               NaN  
3                         NaN            1624.0             658.0  
4                         NaN               NaN               NaN  

[5 rows x 57 columns]

## ユニークな値を取り除いたデータフレーム

In [7]:
print(df_cleaned.nunique(dropna=False).to_string())

Unnamed: 0                             17492
Recording timestamp                    17479
Computer timestamp                     17479
Sensor                                     3
Eyetracker timestamp                   17312
Event                                      9
Event value                                9
Gaze point X                            1997
Gaze point Y                            1007
Gaze point left X                       1957
Gaze point left Y                        970
Gaze point right X                      1962
Gaze point right Y                      1004
Gaze direction left X                  11145
Gaze direction left Y                  10394
Gaze direction left Z                   4912
Gaze direction right X                 10947
Gaze direction right Y                 10321
Gaze direction right Z                  5482
Pupil diameter left                     1426
Pupil diameter right                    1335
Pupil diameter filtered                 1402
Validity l

## Sensor事にデータフレームを分割

In [8]:
# NaNを含む行
df_sensor_nan = df_cleaned[df_cleaned['Sensor'].isna()]

# "Mouse" の行
df_sensor_mouse = df_cleaned[df_cleaned['Sensor'] == "Mouse"]

# "Eye Tracker" の行
df_sensor_eye = df_cleaned[df_cleaned['Sensor'] == "Eye Tracker"]


### 問題の既知代わりの時間を取得する

In [9]:
df_sensor_nan

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
0               0                    0           327258000    NaN   
187           187              2883075           330141075    NaN   
6943         6943            117087801           444345801    NaN   
6952         6952            117203696           444461696    NaN   
8286         8286            156132146           483390146    NaN   
8292         8292            156264093           483522093    NaN   
8299         8299            156644481           483902481    NaN   
8300         8300            156644481           483902481    NaN   
8623         8623            161667591           488925591    NaN   
8624         8624            161667591           488925591    NaN   
10497       10497            192828433           520086433    NaN   
10500       10500            192866352           520124352    NaN   
10501       10501            192866352           520124352    NaN   
10802       10802            197857190           525115190    NaN   
10803       10803            197857190           525115190    NaN   
12661       12661            228748261           556006261    NaN   
12664       12664            228780350           556038350    NaN   
12665       12665            228780350           556038350    NaN   
12967       12967            233788595           561046595    NaN   
12968       12968            233788595           561046595    NaN   
14939       14939            266583761           593841761    NaN   
14943       14943            266622153           593880153    NaN   
14944       14944            266622153           593880153    NaN   
15244       15244            271613123           598871123    NaN   
15245       15245            271613123           598871123    NaN   
17182       17182            303990988           631248988    NaN   
17186       17186            304030292           631288292    NaN   
17187       17187            304030292           631288292    NaN   
17488       17488            309030849           636288849    NaN   
17491       17491            309212572           636470572    NaN   

       Eyetracker timestamp                          Event Event value  \
0                       NaN                 RecordingStart         NaN   
187                     NaN  Eye tracker Calibration start         NaN   
6943                    NaN                     MouseEvent  Down, Left   
6952                    NaN                     MouseEvent    Up, Left   
8286                    NaN                     MouseEvent  Down, Left   
8292                    NaN                     MouseEvent    Up, Left   
8299                    NaN    Eye tracker Calibration end         NaN   
8300                    NaN             ImageStimulusStart      blank2   
8623                    NaN               ImageStimulusEnd      blank2   
8624                    NaN             ImageStimulusStart  視線誘導問題.003   
10497                   NaN                  KeyboardEvent       Space   
10500                   NaN               ImageStimulusEnd  視線誘導問題.003   
10501                   NaN             ImageStimulusStart      blank2   
10802                   NaN               ImageStimulusEnd      blank2   
10803                   NaN             ImageStimulusStart  視線誘導問題.002   
12661                   NaN                  KeyboardEvent       Space   
12664                   NaN               ImageStimulusEnd  視線誘導問題.002   
12665                   NaN             ImageStimulusStart      blank2   
12967                   NaN               ImageStimulusEnd      blank2   
12968                   NaN             ImageStimulusStart  視線誘導問題.004   
14939                   NaN                  KeyboardEvent       Space   
14943                   NaN               ImageStimulusEnd  視線誘導問題.004   
14944                   NaN             ImageStimulusStart      blank2   
15244                   NaN               ImageStimulusEnd      blank2   
15245            

In [10]:
# "視線誘導問題." を含む行を抽出
df_filtered = df_sensor_nan[df_sensor_nan["Event value"].astype(str).str.contains("視線誘導問題.")].copy()

# 結果の確認（先頭5行を表示）
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']]
print(df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']])

       Recording timestamp  Computer timestamp Event value
8624             161667591           488925591  視線誘導問題.003
10500            192866352           520124352  視線誘導問題.003
10803            197857190           525115190  視線誘導問題.002
12664            228780350           556038350  視線誘導問題.002
12968            233788595           561046595  視線誘導問題.004
14943            266622153           593880153  視線誘導問題.004
15245            271613123           598871123  視線誘導問題.001
17186            304030292           631288292  視線誘導問題.001


### 各問題の開始時刻
|       |Recording timestamp|  Computer timestamp| Event value|
|-------|-------------------|--------------------|------------|
|8624| 161667591|488925591|視線誘導問題.003|
|10500|192866352|520124352|視線誘導問題.003|
|10803|197857190|525115190|視線誘導問題.002|
|12664|228780350|556038350|視線誘導問題.002|
|12968|233788595|561046595|視線誘導問題.004|
|14943|266622153|593880153|視線誘導問題.004|
|15245|271613123|598871123|視線誘導問題.001|
|17186|304030292|631288292|視線誘導問題.001|

In [11]:
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-5/start-stop-timing.csv")

In [12]:
df_sensor_mouse

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
1              1                 9016           327267016  Mouse   
3              3                24233           327282233  Mouse   
5              5                38437           327296437  Mouse   
7              7                54210           327312210  Mouse   
9              9                70170           327328170  Mouse   
...          ...                  ...                 ...    ...   
8345        8345            157096058           484354058  Mouse   
8347        8347            157112084           484370084  Mouse   
8349        8349            157128053           484386053  Mouse   
8350        8350            157144055           484402055  Mouse   
8352        8352            157152051           484410051  Mouse   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1                      NaN   NaN         NaN           NaN           NaN   
3                      NaN   NaN         NaN           NaN           NaN   
5                      NaN   NaN         NaN           NaN           NaN   
7                      NaN   NaN         NaN           NaN           NaN   
9                      NaN   NaN         NaN           NaN           NaN   
...                    ...   ...         ...           ...           ...   
8345                   NaN   NaN         NaN           NaN           NaN   
8347                   NaN   NaN         NaN           NaN           NaN   
8349                   NaN   NaN         NaN           NaN           NaN   
8350                   NaN   NaN         NaN           NaN           NaN   
8352                   NaN   NaN         NaN           NaN           NaN   

      Gaze point left X  ...  Original Media height  Eye movement type  \
1                   NaN  ...                    NaN       EyesNotFound   
3                   NaN  ...                    NaN       EyesNotFound   
5                   NaN  ...                    NaN       EyesNotFound   
7                   NaN  ...                    NaN       EyesNotFound   
9                   NaN  ...                    NaN       EyesNotFound   
...                 ...  ...                    ...                ...   
8345                NaN  ...                 1080.0           Fixation   
8347                NaN  ...                 1080.0           Fixation   
8349                NaN  ...                 1080.0           Fixation   
8350                NaN  ...                 1080.0           Fixation   
8352                NaN  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
1                         16889.0                      1.0               NaN   
3                         16889.0                      1.0               NaN   
5                         16889.0                      1.0               NaN   
7                         16889.0                      1.0               NaN   
9                         16889.0                      1.0               NaN   
...                           ...                      ...               ...   
8345                        399.0                    156.0             696.0   
8347                        399.0                    156.0             696.0   
8349                        399.0                    156.0             696.0   
8350                        399.0                    156.0             696.0   
8352                        399.0                    156.0             696.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
1                  NaN                         NaN   
3                  NaN                         NaN   
5                  NaN                         NaN   
7                  NaN                         NaN   
9                  NaN                         NaN   
...                ...                         ...   
8345             413.0                      0.36

In [13]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                14166           327272166  Eye Tracker   
4               4                31285           327289285  Eye Tracker   
6               6                47926           327305926  Eye Tracker   
8               8                64343           327322343  Eye Tracker   
10             10                81208           327339208  Eye Tracker   
...           ...                  ...                 ...          ...   
17485       17485            308986112           636244112  Eye Tracker   
17486       17486            309002896           636260896  Eye Tracker   
17487       17487            309019647           636277647  Eye Tracker   
17489       17489            309035985           636293985  Eye Tracker   
17490       17490            309052911           636310911  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2               342651795.0   NaN         NaN           NaN           NaN   
4               342668913.0   NaN         NaN           NaN           NaN   
6               342685553.0   NaN         NaN           NaN           NaN   
8               342701969.0   NaN         NaN           NaN           NaN   
10              342718833.0   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
17485           651604690.0   NaN         NaN        1359.0         296.0   
17486           651621473.0   NaN         NaN           NaN           NaN   
17487           651638223.0   NaN         NaN           NaN           NaN   
17489           651654560.0   NaN         NaN           NaN           NaN   
17490           651671485.0   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
4                    NaN  ...                    NaN       EyesNotFound   
6                    NaN  ...                    NaN       EyesNotFound   
8                    NaN  ...                    NaN       EyesNotFound   
10                   NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
17485             1413.0  ...                 1080.0            Saccade   
17486                NaN  ...                 1080.0       EyesNotFound   
17487                NaN  ...                 1080.0       EyesNotFound   
17489                NaN  ...                    NaN       EyesNotFound   
17490                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                          16889.0                      1.0               NaN   
4                          16889.0                      1.0               NaN   
6                          16889.0                      1.0               NaN   
8                          16889.0                      1.0               NaN   
10                         16889.0                      1.0               NaN   
...                            ...                      ...               ...   
17485                         17.0                    928.0               NaN   
17486                         67.0                    177.0               NaN   
17487                         67.0                    177.0               NaN   
17489                         67.0                    177.0               NaN   
17490                         67.0                    177.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
4                   NaN                         NaN   
6                   NaN                         NaN   
8                   NaN                         NaN   
10                  NaN         

## Mouse以外のデータを抽出

In [14]:
df_cleaned_ver2 = df_cleaned[df_cleaned['Sensor'] != "Mouse"]
nunique_series_ver2 = df_cleaned_ver2.nunique(dropna=False)
single_unique_cols_ver2 = nunique_series_ver2[nunique_series_ver2 == 1].index.tolist()
# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver2:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")

Mouse position X: [nan]
Mouse position Y: [nan]


In [15]:
df_cleaned_ver2 = df_cleaned_ver2.drop(columns=single_unique_cols_ver2)

In [16]:
print(df_cleaned_ver2.nunique(dropna=False).to_string())

Unnamed: 0                             17341
Recording timestamp                    17332
Computer timestamp                     17332
Sensor                                     2
Eyetracker timestamp                   17312
Event                                      9
Event value                                9
Gaze point X                            1997
Gaze point Y                            1007
Gaze point left X                       1957
Gaze point left Y                        970
Gaze point right X                      1962
Gaze point right Y                      1004
Gaze direction left X                  11145
Gaze direction left Y                  10394
Gaze direction left Z                   4912
Gaze direction right X                 10947
Gaze direction right Y                 10321
Gaze direction right Z                  5482
Pupil diameter left                     1426
Pupil diameter right                    1335
Pupil diameter filtered                 1402
Validity l

In [17]:
df_cleaned_ver2.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0           327258000          NaN   
2           2                14166           327272166  Eye Tracker   
4           4                31285           327289285  Eye Tracker   
6           6                47926           327305926  Eye Tracker   
8           8                64343           327322343  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
2           342651795.0             NaN         NaN           NaN   
4           342668913.0             NaN         NaN           NaN   
6           342685553.0             NaN         NaN           NaN   
8           342701969.0             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Presented Media position Y (DACSpx)  \
0           NaN                NaN  ...                                  NaN   
2           NaN                NaN  ...                                  NaN   
4           NaN                NaN  ...                                  NaN   
6           NaN                NaN  ...                                  NaN   
8           NaN                NaN  ...                                  NaN   

   Original Media width  Original Media height  Eye movement type  \
0                   NaN                    NaN                NaN   
2                   NaN                    NaN       EyesNotFound   
4                   NaN                    NaN       EyesNotFound   
6                   NaN                    NaN       EyesNotFound   
8                   NaN                    NaN       EyesNotFound   

   Eye movement event duration  Eye movement type index  Fixation point X  \
0                          NaN                      NaN               NaN   
2                      16889.0                      1.0               NaN   
4                      16889.0                      1.0               NaN   
6                      16889.0                      1.0               NaN   
8                      16889.0                      1.0               NaN   

   Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  
0               NaN                         NaN                         NaN  
2               NaN                         NaN                         NaN  
4               NaN                         NaN                         NaN  
6               NaN                         NaN                         NaN  
8               NaN                         NaN                         NaN  

[5 rows x 55 columns]

In [18]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series_ver3 = df_cleaned_ver2.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols_ver3 = nunique_series_ver3[nunique_series_ver3 == 2].index.tolist()

# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver3:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")


Sensor: [nan 'Eye Tracker']
Presented Media width: [  nan 1920.]
Presented Media height: [  nan 1080.]
Presented Media position X (DACSpx): [nan  0.]
Presented Media position Y (DACSpx): [nan  0.]
Original Media width: [  nan 1920.]
Original Media height: [  nan 1080.]


### Sensor=Mouse以外に着目した後に要素が少ない特徴量について
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Mouse position X| マウス位置 X|`nan`|
|Mouse position Y| マウス位置 Y|`nan`|
|Sensor| センサー|`nan`, `Eye Tracker`|
|Presented Media width|提示されたメディアの幅|`nan` `1920`|
|Presented Media height|提示されたメディアの高さ|`nan`, `1080`|
|Presented Media position X (DACSpx)| 提示されたメディア位置 X|`nan`,  `0`|
|Presented Media position Y (DACSpx)| 提示されたメディアの位置 Y|`nan`,  `0`|
|Original Media width| オリジナルメディアの幅|`nan`, `1920`|
|Original Media height| オリジナルメディアの高さ|`nan`,`1080`|

In [19]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                14166           327272166  Eye Tracker   
4               4                31285           327289285  Eye Tracker   
6               6                47926           327305926  Eye Tracker   
8               8                64343           327322343  Eye Tracker   
10             10                81208           327339208  Eye Tracker   
...           ...                  ...                 ...          ...   
17485       17485            308986112           636244112  Eye Tracker   
17486       17486            309002896           636260896  Eye Tracker   
17487       17487            309019647           636277647  Eye Tracker   
17489       17489            309035985           636293985  Eye Tracker   
17490       17490            309052911           636310911  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2               342651795.0   NaN         NaN           NaN           NaN   
4               342668913.0   NaN         NaN           NaN           NaN   
6               342685553.0   NaN         NaN           NaN           NaN   
8               342701969.0   NaN         NaN           NaN           NaN   
10              342718833.0   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
17485           651604690.0   NaN         NaN        1359.0         296.0   
17486           651621473.0   NaN         NaN           NaN           NaN   
17487           651638223.0   NaN         NaN           NaN           NaN   
17489           651654560.0   NaN         NaN           NaN           NaN   
17490           651671485.0   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
4                    NaN  ...                    NaN       EyesNotFound   
6                    NaN  ...                    NaN       EyesNotFound   
8                    NaN  ...                    NaN       EyesNotFound   
10                   NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
17485             1413.0  ...                 1080.0            Saccade   
17486                NaN  ...                 1080.0       EyesNotFound   
17487                NaN  ...                 1080.0       EyesNotFound   
17489                NaN  ...                    NaN       EyesNotFound   
17490                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                          16889.0                      1.0               NaN   
4                          16889.0                      1.0               NaN   
6                          16889.0                      1.0               NaN   
8                          16889.0                      1.0               NaN   
10                         16889.0                      1.0               NaN   
...                            ...                      ...               ...   
17485                         17.0                    928.0               NaN   
17486                         67.0                    177.0               NaN   
17487                         67.0                    177.0               NaN   
17489                         67.0                    177.0               NaN   
17490                         67.0                    177.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
4                   NaN                         NaN   
6                   NaN                         NaN   
8                   NaN                         NaN   
10                  NaN         

## 各問題毎のデータへ切り分け

In [20]:
# 結果を辞書に格納
extracted_data = {}

targets = df_filtered["Event value"].unique()
for target in targets:
    timestamps = df_filtered[df_filtered["Event value"] == target]["Recording timestamp"].iloc[0:2]
    if len(timestamps) < 2:
        continue  # start/endが不足している場合スキップ

    start, end = timestamps.iloc[0], timestamps.iloc[1]

    extracted = df_sensor_eye[
        (df_sensor_eye["Recording timestamp"] >= start) &
        (df_sensor_eye["Recording timestamp"] <= end)
    ].copy()

    extracted_data[target] = extracted  # 日本語のキーOK

In [21]:
extracted_data["視線誘導問題.002"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
10804       10804            197860705           525118705  Eye Tracker   
10805       10805            197877465           525135465  Eye Tracker   
10806       10806            197894066           525152066  Eye Tracker   
10807       10807            197910624           525168624  Eye Tracker   
10808       10808            197927388           525185388  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
10804           540486121.0   NaN         NaN         574.0         564.0   
10805           540502880.0   NaN         NaN         574.0         555.0   
10806           540519480.0   NaN         NaN         575.0         567.0   
10807           540536037.0   NaN         NaN         578.0         575.0   
10808           540552800.0   NaN         NaN         583.0         576.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
10804              577.0  ...                 1080.0           Fixation   
10805              583.0  ...                 1080.0           Fixation   
10806              572.0  ...                 1080.0           Fixation   
10807              578.0  ...                 1080.0           Fixation   
10808              575.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
10804                        200.0                    289.0             578.0   
10805                        200.0                    289.0             578.0   
10806                        200.0                    289.0             578.0   
10807                        200.0                    289.0             578.0   
10808                        200.0                    289.0             578.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
10804             567.0                      0.3013   
10805             567.0                      0.3013   
10806             567.0                      0.3013   
10807             567.0                      0.3013   
10808             567.0                      0.3013   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
10804                      0.5254               NaN               NaN  
10805                      0.5254               NaN               NaN  
10806                      0.5254               NaN               NaN  
10807                      0.5254               NaN               NaN  
10808                      0.5254               NaN               NaN  

[5 rows x 57 columns]

In [22]:
print(extracted_data["視線誘導問題.002"].nunique(dropna=False).to_string())

Unnamed: 0                             1859
Recording timestamp                    1859
Computer timestamp                     1859
Sensor                                    1
Eyetracker timestamp                   1859
Event                                     1
Event value                               1
Gaze point X                            868
Gaze point Y                            569
Gaze point left X                       866
Gaze point left Y                       589
Gaze point right X                      916
Gaze point right Y                      622
Gaze direction left X                  1825
Gaze direction left Y                  1779
Gaze direction left Z                  1470
Gaze direction right X                 1808
Gaze direction right Y                 1779
Gaze direction right Z                 1444
Pupil diameter left                     487
Pupil diameter right                    450
Pupil diameter filtered                 447
Validity left                   

In [23]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
15246       15246            271614048           598872048  Eye Tracker   
15247       15247            271630518           598888518  Eye Tracker   
15248       15248            271647297           598905297  Eye Tracker   
15249       15249            271663965           598921965  Eye Tracker   
15250       15250            271680391           598938391  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
15246           614234933.0   NaN         NaN        1080.0         301.0   
15247           614251402.0   NaN         NaN        1077.0         308.0   
15248           614268180.0   NaN         NaN        1071.0         315.0   
15249           614284847.0   NaN         NaN        1070.0         319.0   
15250           614301272.0   NaN         NaN        1069.0         324.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
15246             1098.0  ...                 1080.0           Fixation   
15247             1087.0  ...                 1080.0           Fixation   
15248             1090.0  ...                 1080.0           Fixation   
15249             1083.0  ...                 1080.0           Fixation   
15250             1081.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
15246                        333.0                    538.0            1074.0   
15247                        333.0                    538.0            1074.0   
15248                        333.0                    538.0            1074.0   
15249                        333.0                    538.0            1074.0   
15250                        333.0                    538.0            1074.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
15246             334.0                      0.5594   
15247             334.0                      0.5594   
15248             334.0                      0.5594   
15249             334.0                      0.5594   
15250             334.0                      0.5594   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
15246                      0.3088               NaN               NaN  
15247                      0.3088               NaN               NaN  
15248                      0.3088               NaN               NaN  
15249                      0.3088               NaN               NaN  
15250                      0.3088               NaN               NaN  

[5 rows x 57 columns]

In [24]:
extracted_data["視線誘導問題.003"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
8625        8625            161683470           488941470  Eye Tracker   
8626        8626            161699918           488957918  Eye Tracker   
8627        8627            161716715           488974715  Eye Tracker   
8628        8628            161733393           488991393  Eye Tracker   
8629        8629            161749790           489007790  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
8625           504311143.0   NaN         NaN         871.0         403.0   
8626           504327590.0   NaN         NaN         878.0         391.0   
8627           504344386.0   NaN         NaN         876.0         400.0   
8628           504361063.0   NaN         NaN         871.0         404.0   
8629           504377459.0   NaN         NaN         881.0         393.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
8625              900.0  ...                 1080.0           Fixation   
8626              899.0  ...                 1080.0           Fixation   
8627              892.0  ...                 1080.0           Fixation   
8628              891.0  ...                 1080.0           Fixation   
8629              896.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
8625                        266.0                    166.0             876.0   
8626                        266.0                    166.0             876.0   
8627                        266.0                    166.0             876.0   
8628                        266.0                    166.0             876.0   
8629                        266.0                    166.0             876.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
8625             403.0                      0.4564   
8626             403.0                      0.4564   
8627             403.0                      0.4564   
8628             403.0                      0.4564   
8629             403.0                      0.4564   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
8625                      0.3727               NaN               NaN  
8626                      0.3727               NaN               NaN  
8627                      0.3727               NaN               NaN  
8628                      0.3727               NaN               NaN  
8629                      0.3727               NaN               NaN  

[5 rows x 57 columns]

In [25]:
extracted_data["視線誘導問題.004"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
12969       12969            233805221           561063221  Eye Tracker   
12970       12970            233822036           561080036  Eye Tracker   
12971       12971            233838452           561096452  Eye Tracker   
12972       12972            233855139           561113139  Eye Tracker   
12973       12973            233871959           561129959  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
12969           576428434.0   NaN         NaN         600.0         247.0   
12970           576445248.0   NaN         NaN         602.0         237.0   
12971           576461663.0   NaN         NaN         598.0         236.0   
12972           576478349.0   NaN         NaN         597.0         236.0   
12973           576495168.0   NaN         NaN         592.0         241.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
12969              623.0  ...                 1080.0           Fixation   
12970              621.0  ...                 1080.0           Fixation   
12971              623.0  ...                 1080.0           Fixation   
12972              628.0  ...                 1080.0           Fixation   
12973              623.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
12969                        333.0                    408.0             598.0   
12970                        333.0                    408.0             598.0   
12971                        333.0                    408.0             598.0   
12972                        333.0                    408.0             598.0   
12973                        333.0                    408.0             598.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
12969             241.0                      0.3114   
12970             241.0                      0.3114   
12971             241.0                      0.3114   
12972             241.0                      0.3114   
12973             241.0                      0.3114   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
12969                      0.2234               NaN               NaN  
12970                      0.2234               NaN               NaN  
12971                      0.2234               NaN               NaN  
12972                      0.2234               NaN               NaN  
12973                      0.2234               NaN               NaN  

[5 rows x 57 columns]

In [26]:
for key in extracted_data.keys():
    print(f"Key: {key}")
    df = extracted_data[key]
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()
    for col in single_unique_cols:
        unique_val = df[col].unique()
        print(f"  {col}: {unique_val}")

Key: 視線誘導問題.003
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.003']
  Presented Media name: ['視線誘導問題.003.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.002
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.002']
  Presented Media name: ['視線誘導問題.002.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.004
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.004']
  Presen

In [27]:
# 空の辞書に結果を格納
cleaned_data = {}

for key, df in extracted_data.items():
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

    # そのカラムを除外して新しいデータフレームを作成
    df_cleaned = df.drop(columns=single_unique_cols).copy()

    cleaned_data[key] = df_cleaned

    print(f"{key}: Removed {len(single_unique_cols)} columns -> Remaining {df_cleaned.shape[1]} columns")


視線誘導問題.003: Removed 13 columns -> Remaining 44 columns
視線誘導問題.002: Removed 13 columns -> Remaining 44 columns
視線誘導問題.004: Removed 13 columns -> Remaining 44 columns
視線誘導問題.001: Removed 13 columns -> Remaining 44 columns


### カラムの要素が1つしかない変数の説明 写真1
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-5`|
|Recording name| 録画名|`Recording6`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:34:24.211`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:34:24.211`|
|Recording duration| 録画時間|`309213`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`14.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.12`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`51`pixels|
|Average validation accuracy | 平均バリデーション精度|`11`mm|
|Average validation precision SD | 平均バリデーション予測SD|`4.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`5.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.84`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.32`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.43`degrees|
|Average validation accuracy |平均バリデーション精度| `38`pixels|
|Average validation precision SD |平均バリデーション予測SD| `14`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `19`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.001`|
|Presented Media name|提示されたメディア名| `視線誘導問題.001.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [28]:
cleaned_data["視線誘導問題.001"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
15246       15246            271614048           598872048   
15247       15247            271630518           598888518   
15248       15248            271647297           598905297   
15249       15249            271663965           598921965   
15250       15250            271680391           598938391   
...           ...                  ...                 ...   
17180       17180            303960841           631218841   
17181       17181            303977376           631235376   
17183       17183            303993842           631251842   
17184       17184            304010760           631268760   
17185       17185            304027295           631285295   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
15246           614234933.0        1080.0         301.0             1098.0   
15247           614251402.0        1077.0         308.0             1087.0   
15248           614268180.0        1071.0         315.0             1090.0   
15249           614284847.0        1070.0         319.0             1083.0   
15250           614301272.0        1069.0         324.0             1081.0   
...                     ...           ...           ...                ...   
17180           646579728.0        1403.0         770.0             1420.0   
17181           646596262.0        1416.0         771.0             1448.0   
17183           646612727.0        1422.0         750.0             1455.0   
17184           646629643.0        1420.0         747.0             1440.0   
17185           646646177.0        1343.0         647.0             1336.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
15246              309.0              1061.0               294.0  ...   
15247              338.0              1068.0               278.0  ...   
15248              345.0              1052.0               285.0  ...   
15249              353.0              1058.0               286.0  ...   
15250              359.0              1056.0               289.0  ...   
...                  ...                 ...                 ...  ...   
17180              792.0              1386.0               748.0  ...   
17181              797.0              1383.0               745.0  ...   
17183              791.0              1390.0               709.0  ...   
17184              798.0              1399.0               697.0  ...   
17185              680.0              1349.0               615.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
15246                       0.2859                        0.5526   
15247                       0.3133                        0.5561   
15248                       0.3193                        0.5479   
15249                       0.3266                        0.5511   
15250                       0.3325                        0.5501   
...                            ...                           ...   
17180                       0.7336                        0.7219   
17181                       0.7375                        0.7204   
17183                       0.7325                        0.7239   
17184                       0.7386                        0.7288   
17185                       0.6292                        0.7026   

       Gaze point right Y (MCSnorm)  Eye movement type  \
15246                        0.2723           Fixation   
15247                        0.2578           Fixation   
15248                        0.2642           Fixation   
15249                        0.2649           Fixation   
15250                        0.2676           Fixation   
...                             ...                ...   
17180                        0.6930           Fixation   
17181                        0.6897           Fixation   
17183                        0.6560            Saccade   
17184                        0.6456       Unclassif

In [29]:
cleaned_data["視線誘導問題.001"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-5/user-5-pic-1.csv")

### カラムの要素が1つしかない変数の説明 写真4
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-5`|
|Recording name| 録画名|`Recording6`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:34:24.211`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:34:24.211`|
|Recording duration| 録画時間|`309213`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`14.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.12`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`51`pixels|
|Average validation accuracy | 平均バリデーション精度|`11`mm|
|Average validation precision SD | 平均バリデーション予測SD|`4.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`5.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.84`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.32`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.43`degrees|
|Average validation accuracy |平均バリデーション精度| `38`pixels|
|Average validation precision SD |平均バリデーション予測SD| `14`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `19`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.004`|
|Presented Media name|提示されたメディア名| `視線誘導問題.004.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [30]:
cleaned_data["視線誘導問題.004"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
12969       12969            233805221           561063221   
12970       12970            233822036           561080036   
12971       12971            233838452           561096452   
12972       12972            233855139           561113139   
12973       12973            233871959           561129959   
...           ...                  ...                 ...   
14937       14937            266555028           593813028   
14938       14938            266571803           593829803   
14940       14940            266588178           593846178   
14941       14941            266604947           593862947   
14942       14942            266621726           593879726   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
12969           576428434.0         600.0         247.0              623.0   
12970           576445248.0         602.0         237.0              621.0   
12971           576461663.0         598.0         236.0              623.0   
12972           576478349.0         597.0         236.0              628.0   
12973           576495168.0         592.0         241.0              623.0   
...                     ...           ...           ...                ...   
14937           609176229.0        1252.0         520.0             1312.0   
14938           609193003.0        1278.0         526.0             1326.0   
14940           609209377.0        1277.0         532.0             1323.0   
14941           609226145.0        1277.0         531.0             1318.0   
14942           609242923.0        1280.0         523.0             1329.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
12969              234.0               576.0               259.0  ...   
12970              225.0               582.0               249.0  ...   
12971              212.0               574.0               261.0  ...   
12972              212.0               567.0               261.0  ...   
12973              220.0               561.0               262.0  ...   
...                  ...                 ...                 ...  ...   
14937              533.0              1192.0               507.0  ...   
14938              536.0              1229.0               516.0  ...   
14940              545.0              1231.0               518.0  ...   
14941              553.0              1236.0               510.0  ...   
14942              549.0              1230.0               496.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
12969                       0.2166                        0.2999   
12970                       0.2087                        0.3033   
12971                       0.1966                        0.2989   
12972                       0.1959                        0.2953   
12973                       0.2034                        0.2921   
...                            ...                           ...   
14937                       0.4931                        0.6211   
14938                       0.4966                        0.6401   
14940                       0.5049                        0.6414   
14941                       0.5116                        0.6437   
14942                       0.5082                        0.6407   

       Gaze point right Y (MCSnorm)  Eye movement type  \
12969                        0.2401           Fixation   
12970                        0.2304           Fixation   
12971                        0.2412           Fixation   
12972                        0.2419           Fixation   
12973                        0.2423           Fixation   
...                             ...                ...   
14937                        0.4695           Fixation   
14938                        0.4774            Saccade   
14940                        0.4800           Fixation   
14941                        0.4719           Fixat

In [31]:
cleaned_data["視線誘導問題.004"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-5/user-5-pic-4.csv")

### カラムの要素が1つしかない変数の説明 写真2
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-5`|
|Recording name| 録画名|`Recording6`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:34:24.211`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:34:24.211`|
|Recording duration| 録画時間|`309213`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`14.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.12`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`51`pixels|
|Average validation accuracy | 平均バリデーション精度|`11`mm|
|Average validation precision SD | 平均バリデーション予測SD|`4.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`5.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.84`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.32`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.43`degrees|
|Average validation accuracy |平均バリデーション精度| `38`pixels|
|Average validation precision SD |平均バリデーション予測SD| `14`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `19`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.002`|
|Presented Media name|提示されたメディア名| `視線誘導問題.002.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [32]:
cleaned_data["視線誘導問題.002"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
10804       10804            197860705           525118705   
10805       10805            197877465           525135465   
10806       10806            197894066           525152066   
10807       10807            197910624           525168624   
10808       10808            197927388           525185388   
...           ...                  ...                 ...   
12658       12658            228713169           555971169   
12659       12659            228730196           555988196   
12660       12660            228746456           556004456   
12662       12662            228763183           556021183   
12663       12663            228780119           556038119   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
10804           540486121.0         574.0         564.0              577.0   
10805           540502880.0         574.0         555.0              583.0   
10806           540519480.0         575.0         567.0              572.0   
10807           540536037.0         578.0         575.0              578.0   
10808           540552800.0         583.0         576.0              575.0   
...                     ...           ...           ...                ...   
12658           571336686.0         923.0         507.0              941.0   
12659           571353712.0         934.0         504.0              955.0   
12660           571369971.0         929.0         499.0              948.0   
12662           571386697.0         925.0         507.0              939.0   
12663           571403632.0         925.0         504.0              938.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
10804              549.0               572.0               579.0  ...   
10805              534.0               565.0               576.0  ...   
10806              541.0               578.0               594.0  ...   
10807              558.0               577.0               591.0  ...   
10808              555.0               592.0               598.0  ...   
...                  ...                 ...                 ...  ...   
12658              499.0               904.0               516.0  ...   
12659              499.0               914.0               509.0  ...   
12660              498.0               910.0               499.0  ...   
12662              506.0               910.0               509.0  ...   
12663              496.0               912.0               512.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
10804                       0.5082                        0.2978   
10805                       0.4941                        0.2944   
10806                       0.5010                        0.3012   
10807                       0.5170                        0.3007   
10808                       0.5138                        0.3083   
...                            ...                           ...   
12658                       0.4618                        0.4710   
12659                       0.4624                        0.4761   
12660                       0.4611                        0.4740   
12662                       0.4683                        0.4739   
12663                       0.4590                        0.4749   

       Gaze point right Y (MCSnorm)  Eye movement type  \
10804                        0.5364           Fixation   
10805                        0.5329           Fixation   
10806                        0.5496           Fixation   
10807                        0.5477           Fixation   
10808                        0.5537           Fixation   
...                             ...                ...   
12658                        0.4774           Fixation   
12659                        0.4714           Fixation   
12660                        0.4621           Fixation   
12662                        0.4715           Fixat

In [33]:
cleaned_data["視線誘導問題.002"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-5/user-5-pic-2.csv")

### カラムの要素が1つしかない変数の説明 写真3
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-5`|
|Recording name| 録画名|`Recording6`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:34:24.211`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:34:24.211`|
|Recording duration| 録画時間|`309213`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`14.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.12`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`51`pixels|
|Average validation accuracy | 平均バリデーション精度|`11`mm|
|Average validation precision SD | 平均バリデーション予測SD|`4.1`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`5.5`mm|
|Average validation accuracy |平均バリデーション精度| `0.84`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.32`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.43`degrees|
|Average validation accuracy |平均バリデーション精度| `38`pixels|
|Average validation precision SD |平均バリデーション予測SD| `14`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `19`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.003`|
|Presented Media name|提示されたメディア名| `視線誘導問題.003.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [34]:
cleaned_data["視線誘導問題.003"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
8625         8625            161683470           488941470   
8626         8626            161699918           488957918   
8627         8627            161716715           488974715   
8628         8628            161733393           488991393   
8629         8629            161749790           489007790   
...           ...                  ...                 ...   
10494       10494            192785578           520043578   
10495       10495            192802045           520060045   
10496       10496            192818902           520076902   
10498       10498            192835501           520093501   
10499       10499            192852014           520110014   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
8625            504311143.0         871.0         403.0              900.0   
8626            504327590.0         878.0         391.0              899.0   
8627            504344386.0         876.0         400.0              892.0   
8628            504361063.0         871.0         404.0              891.0   
8629            504377459.0         881.0         393.0              896.0   
...                     ...           ...           ...                ...   
10494           535411312.0        1167.0         594.0             1197.0   
10495           535427778.0        1165.0         614.0             1193.0   
10496           535444634.0        1145.0         585.0             1180.0   
10498           535461232.0        1160.0         583.0             1186.0   
10499           535477744.0        1206.0         639.0             1197.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
8625               402.0               842.0               405.0  ...   
8626               398.0               856.0               385.0  ...   
8627               406.0               859.0               394.0  ...   
8628               417.0               852.0               391.0  ...   
8629               396.0               867.0               390.0  ...   
...                  ...                 ...                 ...  ...   
10494              594.0              1136.0               594.0  ...   
10495              601.0              1137.0               627.0  ...   
10496              590.0              1110.0               579.0  ...   
10498              576.0              1135.0               589.0  ...   
10499              589.0              1215.0               689.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
8625                        0.3719                        0.4386   
8626                        0.3684                        0.4460   
8627                        0.3763                        0.4474   
8628                        0.3857                        0.4439   
8629                        0.3667                        0.4514   
...                            ...                           ...   
10494                       0.5495                        0.5916   
10495                       0.5568                        0.5919   
10496                       0.5467                        0.5783   
10498                       0.5336                        0.5909   
10499                       0.5454                        0.6329   

       Gaze point right Y (MCSnorm)  Eye movement type  \
8625                         0.3748           Fixation   
8626                         0.3562           Fixation   
8627                         0.3648           Fixation   
8628                         0.3619           Fixation   
8629                         0.3610           Fixation   
...                             ...                ...   
10494                        0.5496           Fixation   
10495                        0.5804           Fixation   
10496                        0.5363           Fixation   
10498                        0.5454           Fixat

In [35]:
cleaned_data["視線誘導問題.003"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-5/user-5-pic-3.csv")